addons/mail/static/tests/discuss_tests.js

```diff
@@ -2022,5 +2022,32 @@ QUnit.test('no crash on receiving needaction channel message notif with messagin
     discuss.destroy();
 });
 
+QUnit.test('load record form view', async function (assert) {
+    assert.expect(2);
+    const discuss = await createDiscuss({
+        context: {},
+        params: {},
+        data: this.data,
+        services: this.services,
+        session: {
+            user_context: { uid: 99 },
+        },
+        async mockRPC(route, args) {
+            if (args.model === 'my.model' && args.method === 'get_formview_id') {
+                const [resIds, uid] = args.args;
+                assert.deepEqual(resIds, [10],
+                    "should have have called with the correct ID");
+                assert.strictEqual(uid, 99,
+                    "should have have called with the current user");
+            }
+            return this._super(...arguments);
+        },
+    });
+
+    discuss.call('mail_service', 'redirect', 'my.model', 10);
```

Comment:
Should not programatically call a method like this in tests.
Correct way is to display thread with a message that has a link to a non-partner record, and test should click on it.
The link should look something like this: `<a href="#" data-oe-model="my.model" data-oe-id="10">Link</a>`